### ISSUE TITLE
xTrimoPGLM

### ISSUE THREAD

**Issue by JamesStrabala** [JamesStrabala/Capstone-Resorces#18]
### Title
xTrimoPGLM: unified 100-billion-parameterpretrained transformer for deciphering the language of proteins
### URL
PDF uploaded.
### Summary 
xTrimoPGLM (xT) is a protein general language model that uses the xTrimo training methodology for fine-tuning. It uses the general language model framework as a skeleton, and masked training. This model was created to attempt to create a general PLM, which could complete any protein-based task. Unlike in natural language processing, PLMs cannot handle swapping between tasks via encoded labels. They rely on their encoders to process sequences in a way that can interface with task-specific tools. It was trained with 940m sequences, 200b residues, and 100b parameters on 96 NVIDIA machines.
xT outperformed state-of-the-art models at 15/18 fields such as 'protein structure, interactions, functionality and developability'. It also had lower perplexity with ood proteins than said models. It can generate novel sequences with 11% identity and can tune these sequences for specific tasks given supervised tuning.
Training begins with 400m tokens and the masked objective to reduce loss and build understanding. 600m are added as training becomes an 80 (g)-20(m) split to hone representation and generation.
Researchers were able to accurately predict the relationship between loss and the number of parameters used.
1426 proteins were generated starting from the mask that represents the beginning or end of a sequence. Increasing parameter count yielded diminishing returns. Nucleus sampling (selecting from a subset of likely continuations) using different sampling temperature values (randomness coefficient) was used from there until an end token was generated. These sequences were compared to natural proteins in sequence and structure. These sequences were more similar in structure but different and diverse in sequence than ProGen2's output. They were also of lower perplexity. They had an average sequential similarity of 70% and a TM score (confidence in projected fold) of 86%. Most projected structures matched a structure in the UniProt50 database, indicating xT understood the structure.
Supervised Fine Tuning can be used to key xT to generate within a family or function combined with human reinforcement and filtered training data. Proteins generated with these constraints had a higher score for the chosen task than unconstrained generation. They outperformed ProGen2 and ProtGPT2 on most of the tasks.
### Key Points 
* xT is a general PLM, balancing the encoder and decoder in training.
* It has billions of tokens and parameters.
* It generates proteins of high quality and can be tuned to focus on a specific task or family.
### Citation
@article{Chen_Cheng_Li_Geng_Gong_Li_Bei_Tan_Wang_Zeng_et al._2025, title={XTRIMOPGLM: Unified 100-billion-parameter pretrained transformer for deciphering the language of proteins}, DOI={10.1038/s41592-025-02636-z}, journal={Nature Methods}, author={Chen, Bo and Cheng, Xingyi and Li, Pan and Geng, Yangli-ao and Gong, Jing and Li, Shen and Bei, Zhilei and Tan, Xu and Wang, Boyan and Zeng, Xin and et al.}, year={2025}, month={Apr}} 
### Repo link
https://huggingface.co/proteinglm/
